config/config.go

```diff
@@ -1304,3 +1309,32 @@ func getGoGCEnv() int {
 	}
 	return DefaultRuntimeConfig.GoGC
 }
+
+// OTLPConfig is the configuration for writing to the OTLP endpoint.
+type OTLPConfig struct {
+	PromoteResourceAttributes []string `yaml:"promote_resource_attributes,omitempty"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *OTLPConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultOTLPConfig
+	type plain OTLPConfig
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+
+	seen := map[string]struct{}{}
+	i := 0
+	for i < len(c.PromoteResourceAttributes) {
+		s := strings.TrimSpace(c.PromoteResourceAttributes[i])
+		if _, exists := seen[s]; exists {
```

Comment:
@ArthurSens yeah agree, unless the conclusion is we should fail :)

Reply:
Lets do a warning for now. We can revisit this later :)

Reply:
OK will do.

Reply:
@ArthurSens @gouthamve how do I get hold of a logger in `OTLPConfig.UnmarshalYAML`?

Reply:
@ArthurSens @gouthamve since I don't have access to a logger in that method, I decided to return an error instead. I also revised to fail on empty attributes. PTAL.